internal/rbd/mirror.go

```diff
@@ -33,6 +33,39 @@ func (ri *rbdImage) EnableImageMirroring(mode librbd.ImageMirrorMode) error {
 	}
 	defer image.Close()
 
+	parentInfo, err := image.GetParent()
+	if err != nil {
+		return fmt.Errorf("failed to get parent info of image %q: %w",
+			ri, err)
+	}
+
+	if parentInfo.Image.Trash {
+		return fmt.Errorf("failed to enable mirroring on image %q:"+
+			" parent %q is in trash",
+			parentInfo.Image.ImageID, ri)
+	}
+
+	parent, err := ri.getParent()
+	if err != nil {
+		return fmt.Errorf("failed to enable mirroring on image %q:"+
+			" failed to get parent: %w", ri, err)
+	}
+
+	if parent != nil {
+		parentMirroringInfo, err := parent.GetImageMirroringInfo()
+		if err != nil {
+			return fmt.Errorf("failed to enable mirroring on image %q:"+
+				" failed to get mirroring info of parent %q: %w",
+				ri, parent, err)
+		}
+
+		if parentMirroringInfo.State != librbd.MirrorImageEnabled {
+			return fmt.Errorf("failed to enable mirroring on image %q:"+
```

Comment:
we need to return Invalid RPC code when parent in not mirror enabled or when the parent is in trash?

Reply:
https://github.com/csi-addons/spec/tree/main/replication#enablevolumereplication

https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume-errors

So I'll modify csi addons spec EnableVolReplication call errors' INVALID_ARGUEMENT description to include something similar to createvolume_errors  indicating we cannot enable replication on certain volumes.